### PR TITLE
node: stress_object: strip off commas from value strings

### PR DESCRIPTION
**Pull request by bhalevy**
Some resharding_test dtest, e.g. `test_resharding_counter` fails locally for me:
```
    def _run_stress(self, op_cnt, stress_cmd):
        res = self.node.stress_object(stress_cmd)
        if not isinstance(res, dict):
            raise Exception('Error running cassandra-stress: {}'.format(res))
        assert res['total errors'] == 0
>       assert res['total partitions'] >= op_cnt
E       assert 10.0 >= 10000
```

When printing the cassandra-stress stdout I saw that it uses a comma as a 1000s separator:
```
Total partitions          :     10,000 [COUNTER_READ: 10,000]
```

This is probably related to the LOCALE on my machine.

This change explicitly sets the locale to en_US
tha defines comma as the thousands separator
and dot as the decimal point.

The origin local is reset when the function is done.

### CODE REVIEW

**Review by fruch — COMMENTED** [review]
It's not clear why the locale is even involved in this code

I would remove it completely

**Comment by bhalevy** [comment]
> It's not clear why the locale is even involved in this code
> 
> I would remove it completely

I think that the intention was to try to match the locale settings cassandara-stress uses to print its output numbers.
Some locales, like `de_DE` use comma as a decimal point and dot as the thousands separator.

**Comment by bhalevy** [comment]
Yup, setting the locale explicitly to en_US works well as it defines
the thousands separator to comma.

File: ccmlib/node.py [hunk 1]
```diff
@@ -1405,18 +1403,32 @@ def _set_stress_val(key, val, res):
...
-            if start:
-                m = p.match(line)
-                if m:
-                    Node._set_stress_val(m.group(1).strip().lower(), m.group(2).strip(), res)
-            else:
-                if line == 'Results:':
-                    start = True
+        prev_locale, encoding = locale.getlocale(locale.LC_NUMERIC)
+        # Set locale to en_US to simplify printing and parsing of numbers
+        locale.setlocale(locale.LC_NUMERIC, 'en_US')
```
Comment (fruch) [inline ccmlib/node.py:1408]: if we hard code to a specific one, 

we can just replace(',',''), and remove all locale related code

setting the locale like that, might break other code we aren't even aware of, like third party code.

**Comment by fruch** [comment]
@bhalevy this change also introduce 3 more items into the object, which the unittests doesn't know about (and fails)

**Comment by bhalevy** [comment]
> @bhalevy this change also introduce 3 more items into the object, which the unittests doesn't know about (and fails)

I dropped this part in the latest version ([265d8fe](https://github.com/scylladb/scylla-ccm/commit/265d8fe9fb24bf0d4f15f307ff0e0fcf1a654600)).
I thought this could help debug future issues but its not crucial right now.

**Comment by bhalevy** [comment]
Hmm
```
=================================== FAILURES ===================================
________________ TestScyllaRelocatableCluster.test_node_stress _________________

self = <tests.test_scylla_relocatable_cluster.TestScyllaRelocatableCluster object at 0x7fc0b4a16820>
relocatable_cluster = <ccmlib.scylla_cluster.ScyllaCluster object at 0x7fc0b3d2ecd0>

    def test_node_stress(self, relocatable_cluster):
        node1, *_ = relocatable_cluster.nodelist()
        node1: Node
        ret = node1.stress(['write', 'n=10'])
        assert '10 [WRITE: 10]' in ret.stdout
        assert 'END' in ret.stdout
    
>       ret = node1.stress_object(['write', 'n=10'])

tests/test_scylla_relocatable_cluster.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ccmlib/node.py:1408: in stress_object
    locale.setlocale(locale.LC_NUMERIC, 'en_US')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

category = 1, locale = 'en_US'

    def setlocale(category, locale=None):
    
        """ Set the locale for the given category.  The locale can be
            a string, an iterable of two strings (language code and encoding),
            or None.
    
            Iterables are converted to strings using the locale aliasing
            engine.  Locale strings are passed directly to the C lib.
    
            category may be given as one of the LC_* values.
    
        """
        if locale and not isinstance(locale, _builtin_str):
            # convert to string
            locale = normalize(_build_localename(locale))
>       return _setlocale(category, locale)
E       locale.Error: unsupported locale setting

/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/locale.py:608: Error
```

**Comment by fruch** [comment]
> Hmm
> ```
> =================================== FAILURES ===================================
> ________________ TestScyllaRelocatableCluster.test_node_stress _________________
> 
> self = <tests.test_scylla_relocatable_cluster.TestScyllaRelocatableCluster object at 0x7fc0b4a16820>
> relocatable_cluster = <ccmlib.scylla_cluster.ScyllaCluster object at 0x7fc0b3d2ecd0>
> 
>     def test_node_stress(self, relocatable_cluster):
>         node1, *_ = relocatable_cluster.nodelist()
>         node1: Node
>         ret = node1.stress(['write', 'n=10'])
>         assert '10 [WRITE: 10]' in ret.stdout
>         assert 'END' in ret.stdout
>     
> >       ret = node1.stress_object(['write', 'n=10'])
> 
> tests/test_scylla_relocatable_cluster.py:36: 
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> ccmlib/node.py:1408: in stress_object
>     locale.setlocale(locale.LC_NUMERIC, 'en_US')
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> 
> category = 1, locale = 'en_US'
> 
>     def setlocale(category, locale=None):
>     
>         """ Set the locale for the given category.  The locale can be
>             a string, an iterable of two strings (language code and encoding),
>             or None.
>     
>             Iterables are converted to strings using the locale aliasing
>             engine.  Locale strings are passed directly to the C lib.
>     
>             category may be given as one of the LC_* values.
>     
>         """
>         if locale and not isinstance(locale, _builtin_str):
>             # convert to string
>             locale = normalize(_build_localename(locale))
> >       return _setlocale(category, locale)
> E       locale.Error: unsupported locale setting
> 
> /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/locale.py:608: Error
> ```
> 

As I said, we don't know which locale are available where we're gonna run, we shouldn't be setting it inside a library code.

**Comment by bhalevy** [comment]
Ok, I'll go back to square one

**Comment by bhalevy** [comment]
@fruch please merge

File: ccmlib/node.py [hunk 2]
```diff
@@ -1381,24 +1381,25 @@ def _set_keyspace_initial_tablets(stress_options: List[str], initial_tablets:int
 
     @staticmethod
     def _set_stress_val(key, val, res):
-        # Set locale to the user's default value (usually specified in the LANG)
-        locale.setlocale(locale.LC_NUMERIC, '')
+        def parse_num(s):
+            return locale.atof(s.replace(',', ''))
```
Comment (fruch) [inline ccmlib/node.py:1385]:
```suggestion
            return float(s.replace(',', ''))
```

Comment (fruch) [reply ccmlib/node.py:1385]: we don't need to use locale reading a float.

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**